Orange/preprocess/impute.py

```diff
@@ -99,18 +145,46 @@ def __call__(self, data):
         return column
 
 
-class Model:
-    def __init__(self, learner):
+class Model(BaseImputeMethod):
+    name = "Model-based imputer"
+    short_name = "model"
+    description = ""
+
+    def __init__(self, learner=None):
         self.learner = learner
 
-    def __call__(self, data, variable):
+    def __call__(self, data, variable, copy=False):
         variable = data.domain[variable]
         domain = domain_with_class_var(data.domain, variable)
         data = data.from_table(domain, data)
-        model = self.learner(data)
-        assert model.domain.class_var == variable
-        return variable.copy(
-            compute_value=ReplaceUnknownsModel(variable, model))
+
+        if self.learner.check_learner_adequacy(domain):
+            model = self.learner(data)
+            assert model.domain.class_var == variable
+            return variable.copy(
+                compute_value=ReplaceUnknownsModel(variable, model))
+        else:
+            return variable.copy()
```

Comment:
I still do not like this. If I call an instance of a Model imputer with data, I would like to have imputed data. If the learner does not support variable type and variable contains missing values, this should IMO raise an exception.

Or maybe model could accept multiple learners as a parameter, one for continuous and one for discrete?